core/bcast/recast.go

```diff
@@ -69,21 +72,62 @@ func (r *Recaster) Store(_ context.Context, duty core.Duty,
 	return nil
 }
 
+// StorePregen stores pre-generated aggregate signed duty registrations for rebroadcasting.
+func (r *Recaster) StorePregen(_ context.Context, duty core.Duty,
+	pubkey core.PubKey, aggData core.SignedData,
+) error {
+	if duty.Type != core.DutyBuilderRegistration {
+		return nil
+	}
+
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	tuple, ok := r.pregenTuples[pubkey]
+	if ok && tuple.duty.Slot >= duty.Slot {
```

Comment:
i think this should be
```suggestion
	if ok || tuple.duty.Slot >= duty.Slot {
```
instead ?